merlin-server/constraints-dsl-compiler/src/libs/constraints-ast.ts

```diff
@@ -1,38 +1,57 @@
 export enum NodeKind {
-  DummyConstraint = 'DummyConstraint',
-  ConstraintAnd = 'ConstraintAnd',
-  ConstraintOr = 'ConstraintOr'
+  ViolationsOf = 'ViolationsOf',
+  WindowsExpressionAnd = 'WindowsExpressionAnd',
+  WindowsExpressionOr = 'WindowsExpressionOr',
+  WindowsExpressionTrue = 'WindowsExpressionTrue'
 }
 
-export type Constraint = | DummyConstraint;
+/**
+ * Top-level constraints that directly produce violations
+ * when evaluated.
+ */
+export type Constraint = | ViolationsOf;
```

Comment:
Based on a conversation in the huddle last week, I think we established that constraints are allowed to be bare windows expressions.

```suggestion
export type Constraint =
  | ViolationsOf
  | WindowsExpression;
```

I'm not sure what implications this has for serialization, since WindowsExpressions don't have `__astnode`.

Reply:
Agreed. I haven't yet worked out how to do this, but that will be part of the next PR.